pallets/msa/src/rpc/src/tests/mod.rs

```diff
@@ -175,11 +178,58 @@ async fn get_granted_schemas_by_msa_id_with_no_delegation() {
 #[tokio::test]
 async fn get_granted_schemas_by_msa_id_with_bad_provider_id() {
 	let client = Arc::new(TestApi {});
-	let api = MsaHandler::<TestApi, Block>::new(client);
+	let api = MsaHandler::<TestApi, Block, TestPersistentOffchainDB>::new(client, None);
 
 	let result = api.get_granted_schemas_by_msa_id(DELEGATE_A, ProviderId(NOT_EXIST_MSA));
 
 	assert_eq!(true, result.is_ok());
 	let response = result.unwrap();
 	assert!(response.is_none());
 }
+
+#[tokio::test]
+async fn get_keys_by_msa_id_with_disabled_offchain_should_fail() {
+	let client = Arc::new(TestApi {});
+	let api = MsaHandler::<TestApi, Block, TestPersistentOffchainDB>::new(client, None);
+
+	let result = api.get_keys_by_msa_id(NOT_EXIST_MSA);
+
+	assert_eq!(true, result.is_err());
+	assert_eq!("Custom error: OffchainIndexingNotEnabled", result.unwrap_err().to_string());
+}
+
+#[tokio::test]
+async fn get_keys_by_msa_id_with_empty_value_should_work() {
+	let client = Arc::new(TestApi {});
+	let api = MsaHandler::<TestApi, Block, TestPersistentOffchainDB>::new(
+		client,
+		Some(TestPersistentOffchainDB::new()),
+	);
+
+	let result = api.get_keys_by_msa_id(NOT_EXIST_MSA);
+
+	assert_eq!(true, result.is_ok());
+	let response = result.unwrap();
+	assert_eq!(None, response);
+}
+
+#[tokio::test]
+async fn get_keys_by_msa_id_with_value_should_work() {
```

Comment:
nit: would prefer this test retrieving multiple account ids for an MSA, even though you check that those are all pushed to the db in the offchain_storage test.
```
#[tokio::test]
async fn get_keys_by_msa_id_with_value_should_work() {
	let msa_id: MessageSourceId = 10;
	let accounts = vec![Public([1u8; 32]), Public([2u8; 32]), Public([5u8; 32])];
	let client = Arc::new(TestApi {});
	let mut db = TestPersistentOffchainDB::new();
	db.set(
		sp_offchain::STORAGE_PREFIX,
		&get_msa_account_storage_key_name(msa_id),
		&accounts.encode(),
	);
	let api = MsaHandler::<TestApi, Block, TestPersistentOffchainDB>::new(client, Some(db));

	let result = api.get_keys_by_msa_id(msa_id);

	assert_eq!(true, result.is_ok());
	let response = result.unwrap();
	assert_eq!(true, response.is_some());
	let expected_response = vec![
		KeyInfoResponse { msa_id, key: accounts[0] },
		KeyInfoResponse { msa_id, key: accounts[1] },
		KeyInfoResponse { msa_id, key: accounts[2] },
	];
	assert_eq!(, response.unwrap());
}
```